manual/development-process/access-an-environment-db.md

```diff
@@ -0,0 +1,101 @@
+
+   Run this `aws ec2 describe-instances`
+
+   It will print out a JSON like this:
+
+       "Reservations": [
+           {
+               "Groups": [],
+               "Instances": [
+                   {
+                       "AmiLaunchIndex": x,
+                       "ImageId": "xxxxxxxx",
+                       "InstanceId": "xxxxxxxxx",
+       ...etc
+
+Copy the value of the `InstanceId`
+
+###Via Console
+
+Alternatively, you can also get it from the console:
+
+- In the search bar, type in "EC2" 
+- Under "Resources", click "Instances"
+- Find the bastion ec2 instance, for e.g "bastion-ec2-instance-intg" for the intg environment
+- Copy the "InstanceId"
+
+##3. Connecting via Session Manager
+Now you need to connect via the session manager by running the command
+
+   `aws ssm start-session --target <the InstanceId you've copied>`
+   (replacing the angle bracket parameter, with the InstanceId)
+
+##4. Starting PSQL
+
+There is a script in the home directory which allows you to connect to postgres using IAM DB authentication
+```
+cd
+./connect.sh
+```
+
+The script downloads the RDS public certificate if not already there, assumes a role which is allowed to connect to the database, generates a temporary password and uses this to connect to the database. 
+
+##5. Setting up an ssh tunnel
+This assumes you've added the ssh key when [creating the bastion](./applying-or-destroying-a-bastion-host.md#applying-a-bastion-host)
+
+* Add this to your ssh config.
+```
+# SSH over Session Manager
+host i-* mi-*
+    ProxyCommand sh -c "aws ssm start-session --target %h --document-name AWS-StartSSHSession --parameters 'portNumber=%p'"
+```
+* Get the instance id from the instances page in the console or by running
+  `aws ec2 describe-instances --filters Name=instance-state-name,Values=running Name=tag:Name,Values=bastion-ec2-instance-intg`
+
+* Get the database endpoint. There are three ways:
+
+  You can get this from the AWS console by going to RDS, click DB Instances, choose the reader instance from the consignment api database and copy the endpoint.
+
+  You can call `aws rds describe-db-instances` and look for a field called `Address` for the consignment api.
+
+  You can open the `/home/ssm-user/connech.sh` script on the bastion host and the endpoint is in there assigned to the RDSHOST variable.
+* Run the ssh tunnel
+
+`ssh ec2-user@instance_id -N -L 65432:db_host_name:5432`
+
+* Get the cluster endpoint. There are two ways:
+  Select the cluster in the RDS Databases page in the console  
+  Run `aws rds describe-db-cluster-endpoints | jq '.DBClusterEndpoints[] | select(.EndpointType == "READER") | .Endpoint'
+  ` and select the endpoint for the consginment API.
+* Update your hosts file. In *nix systems, this is in `/etc/hosts`, on Windows, it is in `C:\Windows\System32\drivers\etc\hosts` You will need to add an entry like
+
+`127.0.0.1    cluster_endpoint `
+* Get the password for the database
```

Comment:
Probably worth mentioning that this password has a 15 minute expiry time, so people aren't confused when it stops working.

Reply:
Yep, I've added that in as well.